.github/workflows/dotnet.yml

```diff
@@ -35,13 +35,50 @@ jobs:
       if: matrix.os == 'windows-latest'
       shell: bash
       run: dotnet publish -c Release -r win-x64 --self-contained Refresher //p:Version=${VERSION}
+
+    - name: Publish for macOS x64
+      if: matrix.os == 'macos-latest'
+      shell: bash
+      run: dotnet publish -c Release -r osx-x64 --self-contained Refresher /p:Version=${VERSION}
+
+    - name: Publish for macOS ARM64
+      if: matrix.os == 'macos-latest'
+      shell: bash
+      run: dotnet publish -c Release -r osx-arm64 --self-contained Refresher /p:Version=${VERSION}
+
+    - name: Remove duplicate .app for macOS x64
+      if: matrix.os == 'macos-latest'
+      shell: bash
+      working-directory: Refresher/bin/Release/net8.0/osx-x64/publish/
+      run: rm -rfv ./*.app/Contents/MacOS/*.app && codesign -fs - --deep *.app
+
+    - name: Remove duplicate .app for macOS ARM64
+      if: matrix.os == 'macos-latest'
+      shell: bash
+      working-directory: Refresher/bin/Release/net8.0/osx-arm64/publish/
+      run: rm -rfv ./*.app/Contents/MacOS/*.app && codesign -fs - --deep *.app
+      
+    # We need to tarball our macOS and Linux builds, since the ZIP files created by upload-artifact do not retain extended unix file permissions, 
+    # which means the executable bit is lost, which is un-ideal for end users, who will hit weird errors (especially relating to code signing on macOS)
+    - name: 'Tar macOS x64 build'
+      if: matrix.os == 'macos-latest'
+      working-directory: Refresher/bin/Release/net8.0/osx-x64/publish/
+      run: tar -cvf ../../../../../../Refresher-x64.tar *.app
+    - name: 'Tar macOS ARM64 build'
+      if: matrix.os == 'macos-latest'
+      working-directory: Refresher/bin/Release/net8.0/osx-arm64/publish/
+      run: tar -cvf ../../../../../../Refresher-arm64.tar *.app 
+    - name: 'Tar Linux x64'
+      if: matrix.os == 'ubuntu-latest'
+      working-directory: Refresher/bin/Release/net8.0/linux-x64/publish/
+      run: tar -cvf ../../../../../../Refresher.tar * 
```

Comment:
Why aren't we compressing any of these? `-czvf` will make a .tar.gz file for free.

Should also include platform in the final output, e.g. `Refresher-macos-arm64.tar.gz` or `Refresher-linux-x64.tar.gz`

Reply:
github compresses these for us into `.zip`, so its `.tar.zip` for the macos/linux, and just .zip for windows

Reply:
My point still remains... why not just use .tar.gz? That doesn't really make sense. I imagine that making it annoying to extract; since .zip is also a container instead of being purely compression.

You're putting an archive in an archive. That's just silly.

Reply:
when it comes to standard release artifacts (which is what we are dealing with on this particular CI job)
to my knowledge, we have no choice
its .zip only with github

and with MacOS, it actually autodetects .tar.zip and will silently decompress both when you open the outer zip file

Reply:
> when it comes to standard release artifacts (which is what we are dealing with on this particular CI job)

If that's the case I don't mind it being used for artifacts, even if it is a bit silly. But that shouldn't apply for the release workflow, because those can support multiple file types including tarballs.

> and with MacOS, it actually autodetects .tar.zip and will silently decompress both when you open the outer zip file

This also won't apply for 99% of Linux archive utilities. Ark, for example, does not do this.

Reply:
Releases could just be uploaded as tarballs. That one is on me for applying the zipping on the release workflow as well - just add compression and you should be good to go for macOS and most Linux distributions.